Changes to get ficus devstack building through packer

shaidar opened the pull request:
Added the following to the packer build of ficus:
- pressed.cfg to get Ubuntu setup
- some initialization and configurations scripts to bootstrap the instance and get salt as masterless minion operational
- general cleanup

shaidar committed:
Changes to get packer build successfully building ficus devstack instance

shaidar committed:
Added the virtualbox.sh script and changed a path

shaidar committed:
Merge branch 'master' into sar/devstack

shaidar committed:
Added new line character

shaidar committed:
Merge branch 'sar/devstack' of https://github.com/mitodl/mitx-devstack into sar/devstack

shaidar committed:
Removed extra lines at end of scripts

blarghmatey commented on http/ubuntu-16.04/preseed.cfg:
```diff
@@ -0,0 +1,109 @@
+#### Contents of the preconfiguration file (for xenial)
+### Localization
+# Preseeding only locale sets language, country and locale.
+d-i debian-installer/locale string en_US
```
Should this be `en_US-UTF8`?

shaidar replied:
Not sure, but I can try it out and change it if that's the case.

shaidar replied:
That seems to work by changing it to en_US.UTF-8

blarghmatey commented on http/ubuntu-16.04/preseed.cfg:
```diff
@@ -0,0 +1,109 @@
+# If you want to force a hostname, regardless of what either the DHCP
+# server returns or what the reverse DNS entry for the IP is, uncomment
+# and adjust the following line.
+d-i netcfg/hostname string mitodlficus
```
Maybe change this to `mitx-devstack`?

shaidar replied:
changed

blarghmatey commented on packer_devstack.json:
```diff
@@ -0,0 +1,73 @@
+  },
+  {
+    "type": "salt-masterless",
+    "local_state_tree": "~/Documents/gitCode/mitodl/salt-ops/salt",
```
This should be a relative path to the packer file. As it is it is dependent on your specific directory structure.

shaidar replied:
changed

blarghmatey reviewed (changes requested):
Overall looks good, just a couple of minor comments.

shaidar committed:
Changes based on feedback

shaidar committed:
Added an inline shell script so that folks can import their git course by running:
course=<course_name> vagrant provision

Also excluded audio as we got some feedback about error trying to run the Vagrantfile.

blarghmatey commented on Vagrantfile:
```diff
@@ -1,12 +1,20 @@
+COURSE=#{ENV['course']}
+sudo chown -R www-data /tmp/edx/$COURSE/
+cd /edx/app/edxapp/edx-platform
+sudo -u www-data /edx/bin/python.edxapp ./manage.py cms --settings=aws import /edx/var/edxapp/data /tmp/edx/$COURSE
```
I think we want to use the import command from lms. It appears that the cms import command is for the contentstore app, whereas the lms import is for courseware...

shaidar replied:
When I tried it using lms, I couldn't get the course to show up in lms or studio even after pre-creating the course in studio. However, by running the above which I found [here](https://openedx.atlassian.net/wiki/spaces/OpenOPS/pages/60227913/Managing+OpenEdX+Tips+and+Tricks), It seems to work as expected (course shows up in both lms and cms) without having to pre-create it.
